doc/Development_Documentation/05_Objects/01_Object_Classes/05_Class_Settings/56_Preview_Generator.md

```diff
@@ -0,0 +1,68 @@
+# Preview Generator
+
+### Summary
+Preview Generators provide a Service to get more control over the preview Tab. They provide an UI to pass additional parameters to a URL-Generator.
+
+Providers need to implement: `\Pimcore\Model\DataObject\ClassDefinition\PreviewGeneratorInterface`
+
+Parameters returned in the `getParams` method will be rendered as ext-js ComboBoxes. Whatever the User chooses will be passed to the `generatePreviewUrl` method.
+
+
+Provide a Preview Generator within the Class settings:
+![Preview Generator Setup](../../../img/preview_generator_1.png)
+
+
+### Sample PreviewProvider Implementation
+```php
+namespace AppBundle\Service\PreviewParamProvider;
+
+class ProductPreviewParamProvider implements \Pimcore\Model\DataObject\ClassDefinition\PreviewGeneratorInterface
+{
+    protected $productLinkGenerator;
+
+    public function __construct(\AppBundle\Service\LinkGenerator\ProductLinkGenerator $productLinkGenerator)
+    {
+        $this->productLinkGenerator = $productLinkGenerator;
```

Comment:
Is it possible to use Preview Generator without Link Generator?

If it is really dependent on LinkGenerator, can't we just use same service/class which then implements the `PreviewGeneratorInterface` and then pass the parameters?
something like,
```
class ProductLinkGenerator extends AbstractProductLinkGenerator implements LinkGeneratorInterface, PreviewGeneratorInterface
{
    /**
     * @param \Pimcore\Model\DataObject\Concrete $object
     * @param array $params
     * @return string
     */
    public function generatePreviewUrl(\Pimcore\Model\DataObject\Concrete $object, array $params): string {
     ...
     }

     /**
     * @param \Pimcore\Model\DataObject\Concrete $object
     *
     * @return array
     */
    public function getParams(\Pimcore\Model\DataObject\Concrete $object): array {
        return [];
    }

Reply:
yes its completely independent. This is just an example. I guess most will use it similar to this, but there are definitely special cases where you would not want to use the LinkGenerator.

Reply:
Gotcha! but when i just used the `Preview Generator` service then preview tab was missing, so we need to fix it.

Reply:
should work now :)